babel_util/recommenders/bibcouple.py

```diff
@@ -39,13 +35,40 @@ def get_next_id():
 
     paper_ids = invert_dict(paper_ids)
 
+    returnString = []
+
     for i in xrange(S.shape[0]):
         row = S.getrow(i).tocoo()
         recs = [(j, v) for j, v in itertools.izip(row.col, row.data)]
         # A paper shouldn't recommend itself
         recs = filter(lambda x: x[0] != i, recs)
         recs.sort(key=lambda x: x[1], reverse=True)
-        for entry in recs[:args.n]:
-            args.outfile.write("{0} {1} {2}\n".format(paper_ids[i],
-                                                      paper_ids[entry[0]],
-                                                      entry[1]))
+        for entry in recs[:numRecs]:
+            sEntry = "{0} {1} {2}\n".format(paper_ids[i], paper_ids[entry[0]], entry[1])
+            if cmdline:
+                outfile.write(sEntry)
+            else:
+                returnString.append(sEntry)
+    if not cmdline:
```

Comment:
If you have to build an in memory string buffer use StringIO, not a list of strings. Also figure out how Luigi will let us write to file, some of these files are quite large and will use up all memory.